source/CMakeLists.txt

```diff
@@ -24,6 +24,54 @@
 # TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
 # MATERIALS OR THE USE OR OTHER DEALINGS IN THE MATERIALS.
 
+set(GRAMMAR_JSON_FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/spirv.core.grammar.json
+  ${CMAKE_CURRENT_SOURCE_DIR}/extinst.glsl.std.450.grammar.json
+  ${CMAKE_CURRENT_SOURCE_DIR}/extinst.opencl.std.grammar.json
+)
+set(GRAMMAR_INC_FILES
+  ${spirv-tools_BINARY_DIR}/core.insts.inc
+  ${spirv-tools_BINARY_DIR}/glsl.std.450.insts.inc
+  ${spirv-tools_BINARY_DIR}/opencl.std.insts.inc
+)
+set(GRAMMAR_PROCESSING_SCRIPT
+  ${spirv-tools_SOURCE_DIR}/utils/generate_grammar_tables.py)
+
+add_custom_command(OUTPUT ${GRAMMAR_INC_FILES}
+  COMMAND ${PYTHON_EXECUTABLE} ${GRAMMAR_PROCESSING_SCRIPT}
+    --spirv-core-grammar=${CMAKE_CURRENT_SOURCE_DIR}/spirv.core.grammar.json
+    --extinst-glsl-grammar=${CMAKE_CURRENT_SOURCE_DIR}/extinst.glsl.std.450.grammar.json
+    --extinst-opencl-grammar=${CMAKE_CURRENT_SOURCE_DIR}/extinst.opencl.std.grammar.json
+    --core-insts-output=${spirv-tools_BINARY_DIR}/core.insts.inc
+    --glsl-insts-output=${spirv-tools_BINARY_DIR}/glsl.std.450.insts.inc
+    --opencl-insts-output=${spirv-tools_BINARY_DIR}/opencl.std.insts.inc
+    --operand-kinds-output=${spirv-tools_BINARY_DIR}/operand.kinds.inc
+  DEPENDS ${GRAMMAR_PROCESSING_SCRIPT} ${GRAMMAR_JSON_FILES}
+  COMMENT "Generate info tables for SPIR-V instructions and operands." VERBATIM)
+
+# The following .cpp files include the above generated .inc files.
+# Add those .inc files as their dependencies.
+#
+# Why using such an awkward way?
+# * If we use add_custom_target() to define a target to generate all .inc files
+#   and let ${SPIRV_TOOLS} depends on that, then we need to run ninja twice
+#   every time the grammar is updated; once for generating the .inc files, the
+#   second time is ninja find that .inc is updated and then rebuild .cpp files.
+# * If we use add_custom_command() with PRE_BUILD, then the grammar processing
+#   script will always run no matter whether the grammar is updated.
+# * add_dependencies() is used to add *target* dependencies to a target.
+# * The following solution only generates .inc files when the script or the
+#   grammar files is updated, and in a single ninja run.
+set_source_files_properties(
+  ${CMAKE_CURRENT_SOURCE_DIR}/opcode.cpp
+  PROPERTIES OBJECT_DEPENDS ${GRAMMAR_INC_FILES})
```

Comment:
I know that `opcode.cpp` doesn't depends on all `.inc` files. But I don't care of putting two more dependencies here.